### PR TITLE
Port to 1.11 (waypoints, colony border, sounds, warehouse restriction)

### PR DESCRIPTION
**Pull request by Raycoms**
# Changes proposed in this pull request:
- Port to 1.11 of the above-mentioned past prs. 

That should be all which isn't up to data in 1.11

Review please

### CODE REVIEW

**Comment by marvin-bitterlich** [comment]
LGTM it is just the port

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/1032/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Review by marvin-bitterlich — CHANGES_REQUESTED** [review]
well, javadoc errors xD

File: src/main/java/com/minecolonies/coremod/colony/ColonyView.java [hunk 1]
```diff
@@ -59,6 +64,14 @@
...
 
+    /**
+     * The Set of waypoints.
+     */
+    private Set<BlockPos> wayPoints = new HashSet<>();
+
+    /**
+     * The overall happiness of the colony.
+     */
     private double overallHappiness = 5;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/ColonyView.java:75]: MAJOR Assign this magic number 5 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

File: src/main/java/com/minecolonies/coremod/sounds/ModSoundEvents.java [hunk 2]
```diff
@@ -5,12 +5,20 @@
...
+
 /**
  * Registering of sound events for our colony.
  */
 public final class ModSoundEvents
 {
     /**
+     * List of sound handlers.
+     */
+    public static final List<AbstractWorkerSounds> handlers = new ArrayList<>();
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/sounds/ModSoundEvents.java:19]: MINOR Make this member "protected". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2386)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 104 issues
* ![MAJOR][MAJOR] 39 major
* ![MINOR][MINOR] 51 minor
* ![INFO][INFO] 14 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/078a8bcba73afb44eaa8617e537ab83e202c0c29/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.221 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L44](https://github.com/Minecolonies/minecolonies/blob/078a8bcba73afb44eaa8617e537ab83e202c0c29/src/main/java/com/minecolonies/coremod/colony/Colony.java#L44): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 43 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L44](https://github.com/Minecolonies/minecolonies/blob/078a8bcba73afb44eaa8617e537ab83e202c0c29/src/main/java/com/minecolonies/coremod/colony/Colony.java#L44): class "Colony" has 83 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L178](https://github.com/Minecolonies/minecolonies/blob/078a8bcba73afb44eaa8617e537ab83e202c0c29/src/main/java/com/minecolonies/coremod/colony/Colony.java#L178): Assign this magic number 5 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L558](https://github.com/Minecolonies/minecolonies/blob/078a8bcba73afb44eaa8617e537ab83e202c0c29/src/main/java/com/minecolonies/coremod/colony/Colony.java#L558): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L1022](https://github.com/Minecolonies/minecolonies/blob/078a8bcba73afb44eaa8617e537ab83e202c0c29/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1022): Assign this magic number 20 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1023](https://github.com/Minecolonies/minecolonies/blob/078a8bcba73afb44eaa8617e537ab83e202c0c29/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1023): Assign this magic number 60 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1147](https://github.com/Minecolonies/minecolonies/blob/078a8bcba73afb44eaa8617e537ab83e202c0c29/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1147): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1149](https://github.com/Minecolonies/minecolonies/blob/078a8bcba73afb44eaa8617e537ab83e202c0c29/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1149): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1151](https://github.com/Minecolonies/minecolonies/blob/078a8bcba73afb44eaa8617e537ab83e202c0c29/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1151): Assign this magic number 128 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by xavierh** [comment]
LGTM

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/1032/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 104 issues
* ![MAJOR][MAJOR] 39 major
* ![MINOR][MINOR] 51 minor
* ![INFO][INFO] 14 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/6047b683dd21a6c3e1544bb7ed58a61f5d78a0d2/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.221 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L44](https://github.com/Minecolonies/minecolonies/blob/6047b683dd21a6c3e1544bb7ed58a61f5d78a0d2/src/main/java/com/minecolonies/coremod/colony/Colony.java#L44): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 43 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L44](https://github.com/Minecolonies/minecolonies/blob/6047b683dd21a6c3e1544bb7ed58a61f5d78a0d2/src/main/java/com/minecolonies/coremod/colony/Colony.java#L44): class "Colony" has 83 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L178](https://github.com/Minecolonies/minecolonies/blob/6047b683dd21a6c3e1544bb7ed58a61f5d78a0d2/src/main/java/com/minecolonies/coremod/colony/Colony.java#L178): Assign this magic number 5 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L558](https://github.com/Minecolonies/minecolonies/blob/6047b683dd21a6c3e1544bb7ed58a61f5d78a0d2/src/main/java/com/minecolonies/coremod/colony/Colony.java#L558): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L1022](https://github.com/Minecolonies/minecolonies/blob/6047b683dd21a6c3e1544bb7ed58a61f5d78a0d2/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1022): Assign this magic number 20 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1023](https://github.com/Minecolonies/minecolonies/blob/6047b683dd21a6c3e1544bb7ed58a61f5d78a0d2/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1023): Assign this magic number 60 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1147](https://github.com/Minecolonies/minecolonies/blob/6047b683dd21a6c3e1544bb7ed58a61f5d78a0d2/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1147): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1149](https://github.com/Minecolonies/minecolonies/blob/6047b683dd21a6c3e1544bb7ed58a61f5d78a0d2/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1149): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1151](https://github.com/Minecolonies/minecolonies/blob/6047b683dd21a6c3e1544bb7ed58a61f5d78a0d2/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1151): Assign this magic number 128 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by marvin-bitterlich — APPROVED** [review]

**Pull request merged**